internal/fs/inode/file.go

```diff
@@ -189,8 +189,9 @@ func (f *FileInode) openReader(ctx context.Context) (io.ReadCloser, error) {
 	rc, err := f.bucket.NewReader(
 		ctx,
 		&gcs.ReadObjectRequest{
```

Comment:
in what flow is this method called? This is where the read is being done: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/gcsx/random_reader.go#L365

Reply:
@vadlakondaswetha the current function is called only from FileInode::ensureContent(). Did you mean the change in the startRead function in internal/gcsx/random_reader.go ? 
Anyhow I don't understand the comment. Is this about whether we need to make any change in the that function? Or whether gzip encoding even be supported in random_reader ? Or something else?

Reply:
Quick search shows FileInode::ensureContent is called in 4 flows:
fileInode::Read() -> called from FileHandle::Read() in else flow. The if block goes to random_reader. ReadAt(). This  is triggered for sequential reads also.
fileInode:Write() 
fileInode::Truncate()
fileInode::cacheEnsureContent


Basically the if block in fileHandle::Read() is missed. Not sure how the manual testing worked ?

Reply:
By the way, I reconfirmed the change. The read of the above file/object failed on the current master branch with the following error

```
{"name":"root","levelname":"ERROR","severity":"ERROR","message":"ReadFile: input/output error, fh.reader.ReadAt: Reader returned 940 too many bytes\n","timestampSeconds":1691476923,"timestampNanos":116343834}
{"name":"root","levelname":"ERROR","severity":"ERROR","message":"fuse: *fuseops.ReadFileOp error: input/output error\n","timestampSeconds":1691476923,"timestampNanos":116423865}
```

Reply:
(*randomReader).startRead is called only from randomReader.ReadAt(). Not sure what do you mean startRead is called but not readAt() function.

Anyways i was looking for the change in startRead() which is done now.

Reply:
> `Not sure what do you mean startRead is called but not readAt() function`

I was looking at the logs the wrong earlier.

> Anyways i was looking for the change in startRead() which is done now.

ok .